R/labels.R

```diff
@@ -100,7 +100,7 @@ var_labels <- function(x, fill = FALSE) {
   # across columns of x
   for (j in theseq) {
     attr(x[[j]], "label") <- if (!is.na(value[j])) {
-      value[j]
+      unname(value[j])
```

Comment:
did you consider `as.vector()` instead? This would strip _all_ the attributes (including names).

Reply:
It is a good suggestion but I think the values when assigned should not have any particular attribute (it was the source of the problem)